src/range_proof.rs

```diff
@@ -1023,8 +1035,20 @@ where
         dynamic_points.push(h_base.clone());
 
         // Perform the final check using precomputation
+        let padding = 2usize
+            .checked_mul(max_statement.generators.bit_length())
+            .ok_or(ProofError::SizeOverflow)?
+            .checked_mul(max_statement.generators.aggregation_factor())
+            .ok_or(ProofError::SizeOverflow)?
+            .checked_sub(max_mn)
+            .ok_or(ProofError::SizeOverflow)?
+            .checked_sub(max_mn)
+            .ok_or(ProofError::SizeOverflow)?;
```

Comment:
I think this and the previous use^ can go into a function
```rust
fn calculate_padding<P: FromUniformBytes + Compressable + Clone + Precomputable>(
    generators: &RangeParameters<P>,
    a_li_len: usize,
    a_ri_len: usize,
) -> Result<usize, ProofError> {
    2usize
        .checked_mul(generators.bit_length())
        .ok_or(ProofError::SizeOverflow)?
        .checked_mul(generators.aggregation_factor())
        .ok_or(ProofError::SizeOverflow)?
        .checked_sub(a_li_len)
        .ok_or(ProofError::SizeOverflow)?
        .checked_sub(a_ri_len)
        .ok_or(ProofError::SizeOverflow)
}
```

Reply:
Done.